ui/components/src/Table/Table.stories.tsx

```diff
@@ -0,0 +1,434 @@
+  },
+};
+
+/**
+ * The `columns` prop is used to configure an array of columns to render in the
+ * table.
+ *
+ * Each column config requires the following properties:
+ * - `accessorKey`: property inside a data object that you want to use to populate cell values in this column.
+ * - `header`: string to display in the header to identify that column.
+ *
+ * Each column config may include the following optional properties:
+ * - `width`: determines the size of the column in the table. It can be set to a number indicating a pixel width or to "auto."
+ * - `cell`: function used to customize rendering of the content in a given column cell. Note that cells have a hardcoded height your customization must work within.
+ */
+export const ColumnConfig: Story = {
+  args: {
+    data: generateMockTableData(100),
+    columns: COLUMNS,
+  },
+  parameters: {
+    // This story is functionally identical to several other ones and mostly
+    // exists as a place to hang documentation, so we don't need another
+    // screenshot for it.
+    happo: false,
+  },
+};
+
+/**
+ * The table has multiple `density` options to support different use cases.
+ * The density modifies styling that impacts the size and space taken up by
+ * content in the table (e.g. font size, padding).
+ */
+export const Density: Story = {
+  args: {
+    height: 300,
+    width: 400,
+    data: generateMockTableData(100),
+    columns: COLUMNS,
+  },
+  argTypes: {
+    data: {
+      // Hide configuration because we are setting these values in this story.
+      density: {
+        disable: true,
+      },
+    },
+  },
+  render: (args) => {
+    return (
+      <Stack spacing={3}>
+        <div>
+          <Typography variant="h3" gutterBottom>
+            standard
+          </Typography>
+          <Table {...args} density="standard" />
+        </div>
+        <div>
+          <Typography variant="h3" gutterBottom>
+            compact
+          </Typography>
+          <Table {...args} density="compact" />
+        </div>
+      </Stack>
+    );
+  },
+};
+
+/**
+ * If you set, `checkboxSelection` to `true`, the table will render checkboxes
+ * in the first column. The state of selected checkboxes is controlled using
+ * the `rowSelection` and `onRowSelectionChanged` props. You may modify the
+ * color of row checkboxes using the `getCheckboxColor` prop.
+ */
+export const CheckboxSelection: Story = {
+  args: {
+    data: generateMockTableData(500),
+    columns: COLUMNS,
+    checkboxSelection: true,
+    getCheckboxColor: (data) => data.color,
+  },
+  argTypes: {
+    // Disabling values managed within the render below.
+    onRowSelectionChange: {
+      table: {
+        disable: true,
+      },
+    },
+    rowSelection: {
+      table: {
+        disable: true,
+      },
+    },
+    checkboxSelection: {
+      table: {
+        disable: true,
+      },
+    },
+  },
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
The rules of hooks get confused by storybook render props because it can't tell they're technically a functional component.